finalProject/highLevelCC.md

```diff
@@ -0,0 +1,50 @@
+
+  I plan to spend the next week and a half working on building up coding muscle memory, while working through the details of this code. My goal is to have a working `encode()` function that I completely understand and that shifts right OR left by the time classes begin.
+
+  Yeah, so after taking some time away, I came back and read through my code comments once more and BINGO! Problem solved. `currentIndex - num` to shift left. I'm still wondering though if you can achieve this with an i--?
+
+```js
+  // declare the function and name it - encode - and add the parameters for the function to take a string and a number
+  function encode(str, num) {
+  // declare variable for string, adding method to turing the string to lowercase letters
+  var lowerCaseStr = str.toLowerCase();
+  // delcare variable for all letters in the alphabet and use the split method to split the alphabet up into individual letters
+  var alphabet = 'abcdefghijklmnopqrstuvwxyz'.split('');
+  // declare variable for the output string
+  var newStr = '';
+  // create loop counter index zero and iterate through the lowerCaseString.length, set increment or decrement
+  for(var i = 0; i < lowerCaseStr.length; i++) {
+    // declare variable for current letter and set it equal to the lowerCaseStr index position
+    var currentLetter = lowerCaseStr[i];
+    // use if-statement to tell the program to continue to iterate if the current letter is an empty space
+    if(currentLetter === '') {
+      newStr += currentLetter;
+      continue;
+    }
+    // This is where I get a little lost. Declare variable for the current index and set it equal to the alphabet index of the current letter
+    var currentIndex = alphabet.indexOf(currentLetter);
+    // Declare new index variable and set it equal to the current index minus a number which will be an argument passed in.  `currentIndex - num` to shift left!
+    var newIndex = currentIndex - num;
+    // Create if-statements that tell the program to wrap around to 0 index position or 26 index position when the newIndex is greater than 25 or less than 0 keeping the iterator rotating through the alphabet when given long strings.
+    if(newIndex > 25) newIndex = newIndex - 26;
+    if(newIndex < 0) newIndex = newIndex + 26;
+    // use an if-statement to tell the program to put the changed to lowercase letters back to uppercase letters.
+    if(str[i] === str[i].toUpperCase()) {
+      newStr += alphabet[newIndex].toUpperCase();
+    }
+    // otherwise just return the new index letter in lowercase
+    else newStr += alphabet[newIndex];
+  }
+  // return the new string of encoded letters
+  return newStr;
+}
+// Expected output FG
+console.log(encode('Hi', 2));
```

Comment:
It is great that you are researching and breaking down this code line by line. Be sure to experiment with changing things around, and making different console logs, to see what variables hold at different points.